### DISCUSSION TITLE
Welcome to Spock Discussions!

### DISCUSSION THREAD

**Discussion by leonard84** [spockframework/spock#1324] [General]
## Welcome 🖖!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
    * For technical/help questions please post them on [stackoverflow](https://stackoverflow.com/questions/tagged/spock), as we already have a large collection of topics there.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by johannesgottschalk** [comment]
Hello everyone,

I have recently been experimenting with the Spock framework and I am wondering about the following: In where statements it's only possible to use @Shared or static variables methods and the like. I had a look at the code and found the line where this is realized for "where:".

What I wonder about what was the reason for this design decision and if it could make sense to disable this check from the "where:".

I am very much looking forward to hear from you and wish you a happy day!

Johannes 🦄

**Reply by Vampire** [reply]
Hi Johannes,

the restriction could easily be lifted of course.

The problem is, that it would not behave as you expect and because of that it is forbidden.

The `where` block is part of the feature method in the DSL, but actually it is run before the tests, as the result of the `where` block actually defines which tests to run (with which parameters to run the feature). Then per iteration an own instance of the specification is created to run that iteration.

It's not how it technically works, but think of the `where` block as standalone static method that is called before the tests run.

In reality the where block is multiple instance methods that are run on an instance of the specification that is created separately from the ones for running the iterations, so if you would access instance fields, then you would access the fields of that instance which do not help you in any way.

Due to that only access to static and shared state is allowed, as that's the only thing you can sensibly do. And actually you should generally avoid using static but always use shared instead.
